optaplanner-wb-screens/optaplanner-wb-guided-rule-editor/optaplanner-wb-guided-rule-editor-client/src/main/java/org/optaplanner/workbench/screens/guidedrule/client/widget/ConstraintMatchInputWidgetBlurHandler.java

```diff
@@ -18,20 +18,75 @@
 
 import com.google.gwt.event.dom.client.BlurEvent;
 import com.google.gwt.event.dom.client.BlurHandler;
+import org.jboss.errai.ui.client.local.spi.TranslationService;
+
+import static org.optaplanner.workbench.screens.guidedrule.client.resources.i18n.GuidedRuleEditorConstants.ConstraintMatchInputWidgetBlurHandler_DoubleValueParsingError;
+import static org.optaplanner.workbench.screens.guidedrule.client.resources.i18n.GuidedRuleEditorConstants.ConstraintMatchInputWidgetBlurHandler_EmptyValuesAreNotAllowedForModifyScore;
+import static org.optaplanner.workbench.screens.guidedrule.client.resources.i18n.GuidedRuleEditorConstants.ConstraintMatchInputWidgetBlurHandler_IntegerValueParsingError;
+import static org.optaplanner.workbench.screens.guidedrule.client.resources.i18n.GuidedRuleEditorConstants.ConstraintMatchInputWidgetBlurHandler_LongValueParsingError;
 
 public class ConstraintMatchInputWidgetBlurHandler implements BlurHandler {
 
     private ConstraintMatchInputWidget widget;
+    private TranslationService translationService;
+    private String scoreHolderType;
 
-    public ConstraintMatchInputWidgetBlurHandler(ConstraintMatchInputWidget widget) {
+    public ConstraintMatchInputWidgetBlurHandler(final ConstraintMatchInputWidget widget,
+                                                 final TranslationService translationService,
+                                                 final String scoreHolderType) {
         this.widget = widget;
+        this.translationService = translationService;
+        this.scoreHolderType = scoreHolderType;
     }
 
     @Override
     public void onBlur(BlurEvent event) {
-        if (widget.getConstraintMatchValue() == null || widget.getConstraintMatchValue().trim().isEmpty()) {
-            widget.showEmptyValuesNotAllowedError();
+        String inputValue = widget.getConstraintMatchValue();
+        if (inputValue == null || inputValue.trim().isEmpty()) {
+            widget.showError(translationService.getTranslation(ConstraintMatchInputWidgetBlurHandler_EmptyValuesAreNotAllowedForModifyScore));
         } else {
+            inputValue = inputValue.trim();
+            if (inputValue.matches("-?\\s*\\d+(\\.\\d+)?.*")) {
```

Comment:
So you have validation when the textbox looses focus. Great... and if the value is invalid you get a message. What however clears the invalid value? Or after dismissing the error message can I save with invalid values all the same!?